data/trie/trieStorageManager.go

```diff
@@ -168,6 +169,15 @@ func (tsm *trieStorageManager) CancelPrune(rootHash []byte) {
 
 	log.Trace("trie storage manager cancel prune", "root", rootHash)
 	_, _ = tsm.dbEvictionWaitingList.Evict(rootHash)
+	tsm.removeHashFromPruningBuffer(rootHash)
+}
+
+func (tsm *trieStorageManager) removeHashFromPruningBuffer(rootHash []byte) {
+	for i := range tsm.pruningBuffer {
+		if bytes.Equal(tsm.pruningBuffer[i], rootHash) {
+			tsm.pruningBuffer = append(tsm.pruningBuffer[:i], tsm.pruningBuffer[i+1:]...)
```

Comment:
Could be found twice or more the value of rootHash in the tsm.pruningBuffer? If not, you can do a break after this line, just to avoid the next computations of bytes.Equal for the entire range

Reply:
Added break.